vm/actor/src/builtin/miner/state.rs

```diff
@@ -1005,6 +1025,100 @@ impl MinerInfo {
     }
 }
 
+#[cfg(feature = "json")]
+pub mod json {
+    use super::*;
+    use address::json::AddressJson;
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "PascalCase")]
+    pub struct MinerBaseInfoJson {
+        #[serde(with = "bigint_ser::json::opt")]
+        pub miner_power: Option<TokenAmount>,
+
+        #[serde(with = "bigint_ser::json::opt")]
+        pub network_power: Option<TokenAmount>,
+
+        pub sectors: Vec<SectorInfoJson>,
+
+        #[serde(with = "address::json")]
+        pub worker_key: Address,
+
+        pub sector_size: SectorSize,
+
+        #[serde(with = "beacon::json")]
+        pub prev_beacon_entry: BeaconEntry,
+
+        pub beacon_entries: Vec<BeaconEntryJson>,
+
+        pub eligible_for_mining: bool,
+    }
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "PascalCase")]
+    pub struct MinerInfoJson {
+        pub owner: AddressJson,
+
+        pub worker: AddressJson,
+
+        #[serde(with = "address::json::opt")]
+        pub new_worker: Option<Address>,
+
+        #[serde(with = "address::json::opt_vec")]
+        pub control_addresses: Option<Vec<Address>>, // Must all be ID addresses.
```

Comment:
This should just be a vec?

Reply:
Vec will still serialize to [] need to serialize it to nil

Reply:
Why? The go implementation should be able to deserialize the empty array the same as nil

Reply:
This is the issue I came across when running it against the miner